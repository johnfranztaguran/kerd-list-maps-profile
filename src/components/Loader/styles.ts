import { StyleSheet, ViewStyle } from 'react-native'
import { colors } from 'constant'

interface Styles {
  loader: ViewStyle
}

const styles: Styles = StyleSheet.create<Styles>({
  loader: {
    justifyContent: 'center',
    alignItems: 'center',
    height: '100%',
    width: '100%',
    backgroundColor: colors.solidWhite50
  }
})

export default styles
