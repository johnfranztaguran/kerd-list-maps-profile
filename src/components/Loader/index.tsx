import React from 'react'
import { ActivityIndicator, Modal, View } from 'react-native'
import { testIdProps } from 'utils/testAutomation'
import { strings, themes } from 'constant'
import styles from './styles'

const Container = (): JSX.Element => {
  return (
    <Modal transparent animationType="none" visible>
      <View style={styles.loader}>
        <ActivityIndicator
          {...testIdProps(
            strings.testIdText.prefixView + strings.testIdText.loader
          )}
          size="large"
          color={themes.colors.primary}
        />
      </View>
    </Modal>
  )
}

export default Container
