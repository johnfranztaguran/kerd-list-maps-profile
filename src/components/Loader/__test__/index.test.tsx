import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'
import Loader from '../index'

describe('Component: Loader', () => {
  it('renders correctly', () => {
    const { baseElement } = render(<Loader />)
    expect(baseElement).toMatchSnapshot()
  })
})
