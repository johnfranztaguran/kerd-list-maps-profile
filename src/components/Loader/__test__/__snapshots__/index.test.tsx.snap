// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Loader renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      transparent={true}
      visible={true}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF80",
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <ActivityIndicator
          accessibilityLabel="view-loader"
          accessible={true}
          animating={true}
          color="#38647F"
          hidesWhenStopped={true}
          size="large"
          testID="view-loader"
        />
      </View>
    </Modal>
  </View>
</View>
`;
